### PR TITLE
20429 ee10 ddmodel fat

### LINKED ISSUE #20429: Jakarta EE 10: Enable EE 10 FAT cases
Two EE 10 FAT test classes were disabled, pending necessary enablement.

* open-liberty/dev/com.ibm.ws.javaee.ddmodel_fat/
  * fat/src/com/ibm/ws/javaee/ddmodel/suite/
    * FATSuite
  * fat/src/com/ibm/ws/javaee/ddmodel/suite/core/  
    * DDValidAppTests_J10
    * DDValidModTests_J10

### PR DESCRIPTION
**Pull request by tbitonti**
Test debt: Enable the two new EE 10 ddmodel FAT test classes.

**Review note:** The first two commits are delivered by:
https://github.com/OpenLiberty/open-liberty/issues/20386

The third and fourth commits are "new":
https://github.com/OpenLiberty/open-liberty/pull/20435/commits/7afc2de940bf5e7df077204dbb5dbad12c4361a5
https://github.com/OpenLiberty/open-liberty/pull/20435/commits/942d5c2619a6148ea0a2edab9ad6d31f90f6e52b

fixes #20429

### CODE REVIEW

**Comment by tbitonti** [comment]
closes #20429

**Comment by tbitonti** [comment]
#build

**Comment by tbitonti** [comment]
The update of the features from servlet-5.0 to servlet-6.0 impacts the resolution of the EJB dependencies by the AutoFVT framework.

The AutoFVT framework automatically computes tested features by performing a static analysis of the features provisioned in FAT server configurations.  Explicit features and their dependents are automatically added as tested features.  Conditional "tolerated" features are not automatically added as tested features.

ejb-4.0 remains as a provisioned feature.  That has dependencies:
`ejb-4.0 depends on connectors-2.0 tolerates connectors-2.1`

Also, connectors-2.1 has a dependency on xmlBinding-4.0
`connectors-2.1 depends on xmlBinding-4.0`

Provisioning servlet-6.0 triggers EE 10 dependencies, which includes connectors-2.1.  That is, the EJB dependency changes from connectors-2.0 to connectors-2.1.  Consequently, the dependency on connectors-2.1 triggers the dependency on xmlBinding-4.0.  Since these are conditional dependencies, they are not handled automatically by AutoFVT.

That is, "connectors-2.1" and "xmlBinding-4.0" must be explicitly added to the tested features list:

`tested.features: connectors-2.1, xmlbinding-4.0`

**Comment by tbitonti** [comment]
This test enables two new EE 10 test classes.  See the related issue for a test list.

File: dev/com.ibm.ws.javaee.ddmodel_fat/bnd.bnd [hunk 1]
```diff
@@ -23,6 +23,10 @@ src: \
 
 test.project: true
 
+# Not sure why these in particular must be added.
```
Comment (jhanders34) [inline dev/com.ibm.ws.javaee.ddmodel_fat/bnd.bnd]: I would remove the comment.  If you don't want them you can add connectors-2.1 to the server.xml for the Jakarta EE 10 server xml file.

**Review by jhanders34 — APPROVED** [review]

File: dev/com.ibm.ws.javaee.ddmodel_fat/bnd.bnd (hunk 1 above)
Comment (tbitonti) [reply dev/com.ibm.ws.javaee.ddmodel_fat/bnd.bnd]: I've updated the comment and refreshed the PR.

**Comment by tbitonti** [comment]
#libby
#build

**Comment by tbitonti** [comment]
The single test failure is unrelated:
* com.ibm.ws.microprofile.rest.client_fat FAT tests | 48 | 1 | 0 |   |  
  * com.ibm.ws.microprofile.rest.client.fat.SseTest | 4 | 1 | 0 | 21 s | 15
    * SseClientTestServlet.testPublisherInteger_EE8_FEATURES_MicroProfile_40

**Pull request merged**